### PR TITLE
Add TestRunFilter custom marshalling

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Fixes pagination when a product has a label (or, a label is used in general).

### CODE REVIEW

File: shared/product_spec.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	}
+	return s
+}
+
+func (p ProductSpecs) Len() int           { return len(p) }
+func (p ProductSpecs) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+func (p ProductSpecs) Less(i, j int) bool { return p[i].String() < p[j].String() }
+
+type productSpecNoCustomMarshalling ProductSpec
+type marshallableProductSpec struct {
```
Comment (Hexcles) [inline shared/product_spec.go]: What are these two types for? They don't seem to be referenced anywhere.

File: shared/util.go [hunk 2]
```diff
@@ -90,6 +90,18 @@ func ToStringSlice(set mapset.Set) []string {
 	return result
 }
 
+// ToStringSet converts a typed slice to a set.
+func ToStringSet(slice []string) mapset.Set {
```
Comment (Hexcles) [inline shared/util.go]: This seems the same as `NewSetFromStringSlice`.

File: shared/product_spec.go (hunk 1 above)
Comment (lukebjerring) [reply shared/product_spec.go]: Remnants of making the same change to product spec, only to decide that stringifying and parsing was a better call. Removed.

File: shared/util.go (hunk 2 above)
Comment (lukebjerring) [reply shared/util.go]: Whoops. Looks like my naming choices aren't deterministic.

**Pull request merged**